src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala

```diff
@@ -159,9 +157,15 @@ class TasksResource @Inject() (
       val tasksByAppId: Map[PathId, Seq[Instance]] = maybeInstances.flatten
         .groupBy(instance => instance.instanceId.runSpecId)
         .map { case (appId, instances) => appId -> instances.to[Seq] }(collection.breakOut)
+      val affectedApps = tasksIdToAppId.values.flatMap(appId => groupManager.app(appId))(collection.breakOut)
+      // FIXME (gkleiman): taskKiller.kill a few lines below also checks authorization, but we need to check ALL before
+      // starting to kill tasks
+      affectedApps.foreach(checkAuthorization(UpdateRunSpec, _))
+      val toKill = tasksByAppId
+        .filter { case (appId, _) => affectedApps.exists(app => app.id == appId) }
```

Comment:
This is strange to me. Why do we have to filter at all? If we replace `affectedApps` with its assignment we get

```scala
tasksByAppId.foreach { case (appId, _) =>
  tasksIdToAppId.values
    .flatMap(appId => groupManager.app(appId))
    .exists(app => app.id == appId) }
}
``` 

How do `tasksByAppId` and `tasksIdToAppId` differ? It seems we are just trying to find the apps that are in both but I cannot find how they differ at all.